### PR TITLE
Add highchart for Dataset Explorer 

### PR DESCRIPTION
**Pull request by zhb000**
Add highchart for Dataset Explorer 

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [x] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [x] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [ ] responsibleai/model-assessment

Python packages changed:

- [ ] raiwidgets
- [ ] responsibleai
- [ ] erroranalysis
- [ ] rai_core_flask

## Tests

- [ ] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Screenshots (if appropriate):
Capture

## Documentation:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: libs/dataset-explorer/src/lib/buildScatterTemplate.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { localization } from "@responsible-ai/localization";
+
+export function buildScatterTemplate(
+  jointData: JointDataset,
+  chartProps: IGenericChartProps,
+  x: any,
+  y: any,
+  customData: any
+): string {
+  let hovertemplate = "";
```
Comment (imatiach-msft) [inline libs/dataset-explorer/src/lib/buildScatterTemplate.ts:14]: nit: should this be camelCase, like hoverTemplate?

File: libs/dataset-explorer/src/lib/buildScatterTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const xName = jointData.metaDict[chartProps.xAxis.property].label;
+  const yName = jointData.metaDict[chartProps.yAxis.property].label;
+  if (chartProps.xAxis) {
+    if (chartProps.xAxis.options.dither) {
+      hovertemplate += `${xName}: ${customData.X}<br>`;
+    } else {
+      hovertemplate += `${xName}: ${x}<br>`;
```
Comment (imatiach-msft) [inline libs/dataset-explorer/src/lib/buildScatterTemplate.ts:21]: maybe extract this into a small one line function and re-use it to output the string in the ```key: value<br>``` format?

File: libs/dataset-explorer/src/lib/getDatasetScatterOption.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// Licensed under the MIT License.
+
+import { IGenericChartProps, JointDataset } from "@responsible-ai/core-ui";
+import { IPlotlyProperty } from "@responsible-ai/mlchartlib";
+import _ from "lodash";
+
+import { getDatasetScatter } from "./getDatasetScatter";
+
+export function getDatasetScatterOption(
+  jointData: JointDataset,
```
Comment (imatiach-msft) [inline libs/dataset-explorer/src/lib/getDatasetScatterOption.ts:11]: is there some way to move to the new dataset instead of JointDataset here?

Comment (imatiach-msft) [reply libs/dataset-explorer/src/lib/getDatasetScatterOption.ts:11]: and similarly in getDatasetScatter?

**Review by imatiach-msft — APPROVED** [review]

File: libs/dataset-explorer/src/lib/buildScatterTemplate.ts (hunk 2 above)
Comment (zhb000) [reply libs/dataset-explorer/src/lib/buildScatterTemplate.ts:21]: This is different from plotly, especially for the customized data, will figure out if there is a workaround.

**Pull request merged**